src/tsolvers/egraph/EnodeStore.cc

```diff
@@ -150,9 +150,7 @@ vec<PTRefERefPair> EnodeStore::constructTerm(PTRef tr) {
 }
 
 bool EnodeStore::needsRecursiveDefinition(PTRef tr) const {
-    bool recdef = true;
-    for (auto ch : logic.getPterm(tr)) {
-        recdef &= needsEnode(ch);
-    }
-    return recdef;
+    Pterm const & t = logic.getPterm(tr);
+    opensmt::span const sp(t.begin(), t.size());
```

Comment:
Do we actually need the span here? We could use `t.begin()` and `t.end()` directly in `std::all_of`, no?

Reply:
Fixed.